smartsim/_core/_cli/build.py

```diff
@@ -285,73 +223,236 @@ def build_redis_ai(
+    if not Versioner.ONNX:
+        py_version = sys.version_info
+        msg = (
+            "An onnx wheel is not available for "
+            f"Python {py_version.major}.{py_version.minor}. "
+            "Instead consider using Python 3.8 or 3.9 with RedisAI "
+        )
+        if sys.platform == "linux":
+            msg += "1.2.5 or "
+        msg += "1.2.7."
+        raise SetupError(msg)
+    _confirm_package_in_python_env(
+        {
+            "onnx": Version_(f"{Versioner.ONNX}"),
+            "skl2onnx": Version_(f"{Versioner.REDISAI.skl2onnx}"),
+            "onnxmltools": Version_(f"{Versioner.REDISAI.onnxmltools}"),
+            "scikit-learn": Version_(f"{getattr(Versioner.REDISAI, 'scikit-learn')}"),
+        },
+        install_on_absent=fetch_if_missing,
+        install_on_conflict=False,
+        verbose=verbose,
+    )
+
+
+def install_py_tf_version(
+    fetch_if_missing: bool = False,
+    verbose: bool = False,
+) -> None:
+    """Check Python environment for a compatible TensorFlow installation"""
+    logger.info("Searching for a compatible TF install...")
+    _confirm_package_in_python_env(
+        {"tensorflow": Versioner.TENSORFLOW},
+        install_on_absent=fetch_if_missing,
+        install_on_conflict=False,
+        verbose=verbose,
+    )
+
+
+def _confirm_package_in_python_env(
+    packages: t.Mapping[str, t.Optional[Version_]],
+    package_pinning: _TPinningStr = "==",
+    end_point: t.Optional[str] = None,
+    validate_installed_version: t.Optional[
+        t.Callable[[str, t.Optional[Version_]], bool]
+    ] = None,
+    install_on_absent: bool = False,
+    install_on_conflict: bool = False,
+    verbose: bool = False,
+) -> None:
+    # TODO: Do not like how the defualt validation function will always look for
+    #       a `==` pinning. Maybe turn `BuildEnv.check_installed` into a factory
+    #       that takes a pinning and an appropiate validation fn?
+    validate_installed_version = validate_installed_version or BuildEnv.check_installed
+    missing, conflicts, to_install, to_uninstall = _assess_python_env(
+        packages,
+        package_pinning,
+        validate_installed_version,
+        install_on_absent,
+        install_on_conflict,
+        verbose=verbose,
+    )
+
+    if missing or conflicts:
+        indent = "\n\t"
+        fmt_list: t.Callable = lambda n, l: f"{n}:{indent}{indent.join(l)}" if l else ""
+        missing_str = fmt_list("Missing", missing)
+        conflict_str = fmt_list("Conflicting", conflicts)
+        sep = "\n" if missing_str and conflict_str else ""
+        logger.warning(
+            "Python Env Status Warning!\n"
+            "Requested Packages are Missing or Conflicting:\n\n"
+            f"{missing_str}{sep}{conflict_str}"
+            "\n\nConsider installing packages at the requested versions via "
+            "`pip` or re-running `smart build` with `--modify_python_env`"
+        )
+    if to_uninstall:
```

Comment:
Does taking the install & uninstall actions belong in a method called `_confirm_package_in_python_env`?

Reply:
Good point! Will change!